pkg/reconciler/sequence/sequence.go

```diff
@@ -224,3 +232,87 @@ func (r *Reconciler) trackAndFetchChannel(ctx context.Context, seq *v1.Sequence,
 	}
 	return obj, err
 }
+
+func (r *Reconciler) removeUnwantedChannels(ctx context.Context, channelResourceInterface dynamic.ResourceInterface, seq *v1.Sequence, wanted []*eventingduckv1.Channelable) error {
+	channelObjRef := corev1.ObjectReference{
+		Kind:       seq.Spec.ChannelTemplate.Kind,
+		APIVersion: seq.Spec.ChannelTemplate.APIVersion,
+	}
+
+	l, err := r.channelableTracker.ListerFor(channelObjRef)
+	if err != nil {
+		logging.FromContext(ctx).Errorw("Error getting lister for Channel", zap.Any("channelRef", channelObjRef), zap.Error(err))
+		return err
+	}
+
+	exists, err := l.ByNamespace(seq.GetNamespace()).List(labels.Everything())
+	if err != nil {
+		logging.FromContext(ctx).Errorw("Error listing Channels", zap.Any("namespace", seq.Namespace), zap.Any("channelRef", channelObjRef), zap.Error(err))
+		return err
+	}
+
+	for _, c := range exists {
+		channelable, ok := c.(*eventingduckv1.Channelable)
+		if !ok {
+			logging.FromContext(ctx).Errorw("Failed to convert to Channelable Object", zap.Any("channel", c), zap.Error(err))
+			return fmt.Errorf("failed to convert to Channelable Object: %+v", c)
+		}
+
+		if !channelable.GetDeletionTimestamp().IsZero() ||
+			!metav1.IsControlledBy(channelable, seq) {
+			continue
+		}
+
+		used := false
+		for _, ch := range wanted {
+			if equality.Semantic.DeepDerivative(ch, channelable) {
```

Comment:
DeepDerivative seems overkill. Why not just comparing the names? Bonus: no need to cast to Channelable.

Reply:
actually DeepDerivative may be wrong. Channelable might be an old Channel (Lister contains cached objects) so DeepDerivative might returns false when the channel has been updated with new values (like a new subscriber). Just comparing the names is IMO safer.

Reply:
Done at https://github.com/knative/eventing/pull/5718/commits/ad8aee54d965bf527d74fa76246d25b5d589a979